velox/functions/sparksql/tests/ArithmeticTest.cpp

```diff
@@ -133,6 +138,41 @@ TEST_F(RemainderTest, int64) {
   EXPECT_EQ(-1, remainder<int64_t>(INT64_MIN, INT64_MAX));
 }
 
+TEST_F(RemainderTest, double) {
+  constexpr double kInf = std::numeric_limits<double>::infinity();
+  constexpr double kNan = std::numeric_limits<double>::quiet_NaN();
```

Comment:
Consider adding a local function and removing remainder_value as it is used only for double and float.
> const auto remainderDouble = [&](std::optional\<double> a, std::optional\<double> b) {
  return remainder<double>(a, b).value();
}